lib/AuthProvider.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:http/http.dart';
+
+import 'app_config.dart';
+
+Future<String> auth(String baseUrl, String token, String email) async {
+  String endpoint = baseUrl + '/auth';
+  Map<String, dynamic> requestBody = {
+    "token": token,
+    "email": email,
+    "clientID":
+        "241748771473-0r3v31qcthi2kj09e5qk96mhsm5omrvr.apps.googleusercontent.com",
+    "table": "Drivers"
+  };
+  return post(endpoint, body: requestBody).then((res) {
+    return res.body;
+  });
+}
+
+Future<String> tokenFromAccount(GoogleSignInAccount account) async {
+  GoogleSignInAuthentication auth;
+  try {
+    auth = await account.authentication;
+    print('okay');
+  } catch (error) {
+    return null;
+  }
+  return auth.idToken;
+}
+
+class AuthProvider with ChangeNotifier {
+  String id;
+  StreamSubscription _userAuthSub;
+  GoogleSignIn googleSignIn;
+
+  AuthProvider(BuildContext context) {
```

Comment:
If having an argument on your provider actually works, then keep this. I thought I had read that having arguments to a provider is not good practice. But if this works and it allows you to access AppConfig.of(context) within the AuthProvider, then keep it and consider adding it to Carriage Rider as well.

Reply:
Seems to work fine.  I think it's reasonable at least to use context as an argument, since ChangeNotifierProvider's 'create' property takes it as an argument anyway. I'll make this change to the Rider app too.